code/modules/mechs/mech_movement.dm

```diff
@@ -6,8 +6,20 @@
 
 /mob/living/exosuit/Move()
 	. = ..()
-	if(. && !istype(loc, /turf/space))
-		playsound(src.loc, mech_step_sound, 40, 1)
+	if(.)
+		if(!istype(loc, /turf/space))
+			playsound(src.loc, mech_step_sound, 40, 1)
+
+		var/turf/T = src.loc
+		if (!isturf(T))	return
+
+		var/turf/B = GetAbove(T)
```

Comment:
You don't seem to check `HasAbove()` anywhere in this block.

Reply:
`GetAbove()` implies `HasAbove()`, checking both is redundant.